### ISSUE TITLE
Collapse and decollapse icon still shown even though there's no child in the drawer

### ISSUE THREAD

**Issue by ReDarkTechnology** [yasirkula/UnityRuntimeInspector#57]
**Description of the bug**

I used the GameObjectFilter feature in RuntimeHierarchy to hide billboard objects with this script
```cs
hierarchy.GameObjectFilter += (Transform obj) => {
    return obj.GetComponent<BillboardSprite>() == null;
};
```
But it still shows the collapse and decollapse icon
![image](https://github.com/yasirkula/UnityRuntimeInspector/assets/53262904/7cbb87da-2e6e-4edd-9b31-d116df5afc98)

**Reproduction steps**
- Create a GameObject and RuntimeHierarchy
- Create a child for the GameObject
- Filter the child of the GameObject so that all of the childs in the GameObject won't be shown in the hierarchy using GameObjectFilter
- Done, collapse and decollapse icon is still shown in the RuntimeHierarchy even though all the childs are filtered

**Platform specs**

Please provide the following info if this is a Unity 3D repository.

- Unity version: *2021.3.33f1*
- Platform: *Android, Windows, Editor*
- Device: *Windows 11, Android 12*
- How did you download the plugin: *GitHub*

**Additional info**
No log.

**Comment by yasirkula** [comment]
Yes this is a known issue. To avoid this issue, at each RuntimeHierarchy refresh, each Transform with children must iterate over all of their children to see if all of them are filtered out. For Transforms with lots of children, my initial assumption is that that'll take considerable amount of time. If you have any ideas, please let me know.

**Comment by ReDarkTechnology** [comment]
Is there a better place for us to discuss this? I still need some help on your asset...

**Comment by yasirkula** [comment]
If you need help about this Issue, we can continue discussing it here. For other topics, you could use this repository's Issues or Discussions.

**Comment by ReDarkTechnology** [comment]
Okay maybe let's just try to solve this issue first

I just want to try a temporary fix for now, I can't exactly locate the function where it would check how many child it should have. It will definitely lag but I can try to just refresh it whenever the scene is changed by the user

**Comment by yasirkula** [comment]
Here, you can use a combination of ChildCount and GetChild to make the necessary controls: https://github.com/yasirkula/UnityRuntimeInspector/blob/b0980bb9a554d8d429cec645593b5c207c49163a/Plugins/RuntimeInspector/Scripts/RuntimeHierarchy/HierarchyData.cs#L54

**Comment by ReDarkTechnology** [comment]
Okay that solves the issue! I just add another abstract property called FilteredChildCount, reimplement the property on all Data types and assign it to CanExpand :D

**Comment by yasirkula** [comment]
Glad to hear it! I thought ChildCount and GetChild could be used to implement this inside HierarchyData.CanExpand directly. Is there a reason you preferred creating an abstract property?

**Comment by yasirkula** [comment]
GetChild and ChildCount are implemented by all different HierarchyData types.

**Comment by ReDarkTechnology** [comment]
Will try to implement them and see if it works well then

**Comment by ReDarkTechnology** [comment]
Reimplemented it like this

```cs
private RuntimeHierarchy _hostHierarchy;
public bool CanExpand { 
	get {
		if (!_hostHierarchy)
			_hostHierarchy = Object.FindObjectOfType<RuntimeHierarchy>(true);

		int count = 0;
		for (int i = 0; i < ChildCount; i++)
		{
            if (_hostHierarchy.GameObjectFilter(GetChild(i)))
                count++;
		}
		return count > 0;
	} 
}
```

**Comment by yasirkula** [comment]
I forgot about having to access RuntimeHierarchy. Still, I think the code can be optimized further as follows: `RuntimeHierarchy _hostHierarchy = Root.Hierarchy;`

**Comment by yasirkula** [comment]
Happy to help. It can remain open for the time being.
